{
  "Severity": "HIGH",
  "UpdatedAt": "2020-08-14T16:09:34Z",
  "Package": {
    "Ecosystem": "PIP",
    "Name": "openapi-python-client"
  },
  "Advisory": {
    "DatabaseId": 2299,
    "Id": "MDE2OlNlY3VyaXR5QWR2aXNvcnlHSFNBLTl4NGMtNjNwZi01MjVm",
    "GhsaId": "GHSA-9x4c-63pf-525f",
    "References": [
      {
        "Url": "https://github.com/triaxtec/openapi-python-client/security/advisories/GHSA-9x4c-63pf-525f"
      },
      {
        "Url": "https://nvd.nist.gov/vuln/detail/CVE-2020-15142"
      },
      {
        "Url": "https://github.com/advisories/GHSA-9x4c-63pf-525f"
      }
    ],
    "Identifiers": [
      {
        "Type": "GHSA",
        "Value": "GHSA-9x4c-63pf-525f"
      },
      {
        "Type": "CVE",
        "Value": "CVE-2020-15142"
      }
    ],
    "Description": "### Impact\nClients generated with a maliciously crafted OpenAPI Document can generate arbitrary Python code. Subsequent execution of this malicious client is arbitrary code execution.\n\nGiving this a CVSS of 8.0 (high) with CVSS:3.0/AV:N/AC:H/PR:L/UI:R/S:C/C:H/I:H/A:H/E:P/RL:U/RC:C .\n\n### Patches\nFix will be included in version 0.5.3\n\n### Workarounds\nInspect OpenAPI documents before generating, or inspect generated code before executing.\n\n### For more information\nIf you have any questions or comments about this advisory:\n* Open an issue in [openapi-python-client](https://github.com/triaxtec/openapi-python-client/issues)\n* Email us at [danthony@triaxtec.com](mailto:danthony@triaxtec.com)",
    "Origin": "UNSPECIFIED",
    "PublishedAt": "2020-08-20T14:38:24Z",
    "Severity": "HIGH",
    "Summary": "Arbitrary Code Generation",
    "UpdatedAt": "2021-01-07T23:41:15Z",
    "WithdrawnAt": ""
  },
  "Versions": [
    {
      "FirstPatchedVersion": {
        "Identifier": ""
      },
      "VulnerableVersionRange": "\u003c 0.5.3"
    }
  ]
}